### PR TITLE
Provide named exports in fork entry points

### PR DESCRIPTION
**Pull request by TrySound**
This adds support for named imports when use node esm. Before because of `module.exports = fork` node could not infer exports so this worked only with bundlers.

For testing run build-release and check entry points.

### CODE REVIEW

**Comment by acywatson** [comment]
> Hey guys! What's your opinion on this solution?

Sorry for the delay - this seems fine? Builds are passing.

The problem is it could theoretically break internal module resolution and it's hard to test that. will look closer later

**Comment by TrySound** [comment]
lexical-history

before

```js
/**
--
* Copyright (c) Meta Platforms, Inc. and affiliates.
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/
'use strict'
const LexicalHistory = process.env.NODE_ENV === 'development' ? require('./LexicalHistory.dev.js') : require('./LexicalHistory.prod.js')
module.exports = LexicalHistory;
```

after

```js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
'use strict'
const LexicalHistory = process.env.NODE_ENV === 'development' ? require('./LexicalHistory.dev.js') : require('./LexicalHistory.prod.js')
exports.createEmptyHistoryState = LexicalHistory.createEmptyHistoryState;
exports.registerHistory = LexicalHistory.registerHistory;
```

lexical-react/useLexicalSubscription

before

```js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
'use strict'
const useLexicalSubscription = process.env.NODE_ENV === 'development' ? require('./useLexicalSubscription.dev.js') : require('./useLexicalSubscription.prod.js')
module.exports = useLexicalSubscription;
```

after

```js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
'use strict'
const useLexicalSubscription = process.env.NODE_ENV === 'development' ? require('./useLexicalSubscription.dev.js') : require('./useLexicalSubscription.prod.js')
module.exports = useLexicalSubscription;
```

**Comment by benjaminsehl** [comment]
If I can help out here at all with any grunt work, let me know.

**Review by acywatson — APPROVED** [review]
I think this works! Gonna wait for a second opinion to merge.

Thanks for the PR!

File: scripts/build.js [hunk 1]
```diff
@@ -592,13 +593,19 @@ async function moveTSDeclarationFilesIntoDist(packageName, outputPath) {
...
 
-function buildForkModule(outputPath, outputFileName) {
+function buildForkModule(outputPath, outputFileName, exports) {
   const lines = [
     getComment(),
     `'use strict'`,
     `const ${outputFileName} = process.env.NODE_ENV === 'development' ? require('./${outputFileName}.dev.js') : require('./${outputFileName}.prod.js')`,
-    `module.exports = ${outputFileName};`,
   ];
+  if (exports.includes('default')) {
```
Comment (fantactuka) [inline scripts/build.js:602]: Any chance we have modules that export both default & named ones?

Comment (acywatson) [reply scripts/build.js:602]: yea good point

**Comment by zhihengGet** [comment]
any update on this

**Comment by ivailop7** [comment]
Superseded by https://github.com/facebook/lexical/pull/5774

**Pull request closed without merging**